README.rst

```diff
@@ -1506,79 +1506,366 @@ Get command details
 
     GET /api/v1/controller/device/{device_id}/command/{command_id}/
 
+List device groups
+##################
+
+.. code-block:: text
+
+    GET /api/v1/controller/group/
+
+Create device group
+###################
+
+.. code-block:: text
+
+    POST /api/v1/controller/group/
+
+Get device group detail
+#######################
+
+.. code-block:: text
+
+    GET /api/v1/controller/group/{id}/
+
+Get device group from certificate common name
+#############################################
+
+.. code-block:: text
+
+    GET /api/v1/controller/cert/{common_name}/group/
+
+This endpoint can be used to retrieve group information and metadata by the
+common name of a certificate used in a VPN client tunnel, this endpoint is
+used in layer 2 tunneling solutions for firewall/captive portals.
+
+It is also possible to filter device group by providing organization slug
+of certificate's organization as show in the example below:
+
+.. code-block:: text
+
+    GET /api/v1/controller/cert/{common_name}/group/?org={org1_slug},{org2_slug}
+
+Get device location
+###################
+
+.. code-block:: text
+
+
+    GET /api/v1/controller/device/{id}/location/
+
+
+Create device location
+######################
+
+.. code-block:: text
+
+    POST /api/v1/controller/device/{id}/location/
+
+You can create ``DeviceLocation`` object by using primary
+keys of existing ``Location`` and ``FloorPlan`` objects as shown in
+the example below.
+
+.. code-block:: json
+
+    {
+        "location": "f0cb5762-3711-4791-95b6-c2f6656249fa",
+        "floorplan": "dfeb6724-aab4-4533-aeab-f7feb6648acd",
+        "indoor": "12.342,23.541"
+    }
+
+.. code-block:: text
+
+    curl -X POST \
+        http://127.0.0.1:8000/api/v1/controller/device/8a85cc23-bad5-4c7e-b9f4-ffe298defb5c/location/ \
+        -H 'content-type: application/json' \
+        -d '{
+            "location": "f0cb5762-3711-4791-95b6-c2f6656249fa",
+            "floorplan": "dfeb6724-aab4-4533-aeab-f7feb6648acd",
+            "indoor": "12.342,23.541"
+            }'
+
+You can also create related ``Location`` and ``FloorPlan`` objects for the
+device directly from this endpoint.
+
+The following example demonstrates creating related location
+object in a single request.
+
+.. code-block:: json
+
+    {
+        "location": {
+            "name": "Via del Corso",
+            "address": "Via del Corso, Roma, Italia",
+            "geometry": "POINT (12.512124 41.898903)",
+            "type": "outdoor",
+        }
+    }
+
+.. code-block:: text
+
+    curl -X POST \
+        http://127.0.0.1:8000/api/v1/controller/device/8a85cc23-bad5-4c7e-b9f4-ffe298defb5c/location/ \
+        -H 'content-type: application/json' \
+        -d '{
+                "location": {
+                    "name": "Via del Corso",
+                    "address": "Via del Corso, Roma, Italia",
+                    "geometry": "POINT (12.512124 41.898903)",
+                    "type": "outdoor"
+                }
+            }'
```

Comment:
The example is missing the authorization header.

Reply:
i intentionally removed the authorization header from the examples since it is not directly related to the information conveyed by the example. i will add them back in all the examples.

Reply:
Thanks for adding it back, I believe it's good to have complete examples so that users can copy/paste them and edit them to do what they need to do.